agents/ecloud/fence_ecloud.py

```diff
@@ -0,0 +1,165 @@
+    del conn
+
+    action = options['--action']
+    vpc = options['--ecloud-vpc']
+    plug = options['--plug']
+
+    url = fence_url(vpc, action, plug)
+    hdrs = headers(options['--password'])
+
+    logging.info("executing '{}' action on '{}'".format(action, plug))
+
+    retries = 0
+    while True:
+        resp = requests.put(url, headers=hdrs)
+        if resp.status_code == 409:
+            # If we attempt to power the instance back on too soon after powering it off,
+            # e.g. during a reboot, the API will return a 409 because while the power status
+            # has changed, the task is still executing. Retry the action until we exceed
+            # retries or get a different status code.
+            if retries >= 6:
+                logging.error("timed out trying to execute '{}' action after repeated 409 codes from API", action)
+                fail(EC_TIMED_OUT)
+
+            time.sleep(2)
+            retries += 1
+            continue
+
+        if resp.status_code != 202:
+            logging.error("unexpected status code '{}' from endpoint '{}': {}".format(
+                resp.status_code, url, resp.text
+            ))
+            
+        break
+
+
+def get_power_fn(conn, options):
+    logging.debug("getting power state")
+    del conn
+
+    vpc = options['--ecloud-vpc']
+    plug = options['--plug']
+
+    url = instance_data_url(vpc, plug)
+    hdrs = headers(options['--password'])
+
+    resp = requests.get(url, headers=hdrs)
+    if resp.status_code != 200:
+        logging.error("unexpected status code ('{}') from endpoint '{}': {}".format(
+            resp.status_code, url, resp.text
+        ))
+        return "bad status {}".format(resp.status_code)
+
+    instance = resp.json()['data']
+    if vpc:
+        logging.debug("power state return value: {}".format(instance['online']))
+        return "on" if instance['online'] else "off"
+    else:
+        if instance['power_status'] == "Online":
+            return "on"
+        elif instance['power_status'] == "Offline":
+            return "off"
+        else:
+            # Could be 'Unknown' or other value
+            return instance['power_status']
+
+
+def headers(password):
+    return {
+        "Authorization": password,
+        "User-Agent": "fence_ecloud"
+    }
+
+
+def itp(url, plug):
+    return url.replace(':ID', plug)
+
+
+def fence_url(vpc, action, plug):
+    if action == "on":
+        return itp(API_VPC_POWER_ON, plug) if vpc else itp(API_V1_POWER_ON, plug)
+    if action == "off":
+        return itp(API_VPC_POWER_OFF, plug) if vpc else itp(API_V1_POWER_OFF, plug)
+    
+    fail_usage("no available API configured for action '{}'".format(action))
+
+
+def instance_data_url(vpc, plug):
+    return itp(API_VPC_INSTANCE_DATA, plug) if vpc else itp(API_V1_INSTANCE_DATA, plug)
+
+
+def main():
+    device_opt = ["passwd", "port", "no_login"]
+
+    all_opt["passwd"]["help"] = "-p, --password=[apikey]        eCloud API Key"
```

Comment:
You might want to use an API key parameter instead like this, and add it to `device_opt[]`:
https://github.com/ClusterLabs/fence-agents/blob/main/agents/ibm_vpc/fence_ibm_vpc.py#L228-L236